### ISSUE TITLE
Issue with style renderer and collision in style key

### ISSUE THREAD

**Issue by savardc** [lokiuz/redraft#28]
When using `createStylesRenderer`, if there's a conflict between 2 overlapping styles, only the last will be applied. 

Example: 
```
UNDERLINE: {
  textDecoration: 'underline'
},
STRIKETHROUGH: {
  textDecoration: 'line-through'
}
```
The rendered text will only have `textDecoration: 'line-through'`

I isolated the problem with the way the styles are merged in `reduceStyles` but my fix involved adding a dependency on [lodash.assignWith](https://www.npmjs.com/package/lodash.assignwith).

Let me know if you're interested in a PR

**Comment by lokiuz** [comment]
Sure PR's are welcome and I don't mind adding dependencies where they make sense. 

Is it supposed to merge any style key or only some of them?

**Comment by savardc** [comment]
So far I've only needed to concatenate values when there was a key conflict. I'm not sure if that handle all cases but it's enough for our use case.

I'll submit a PR so you can have a look

**Comment by lokiuz** [comment]
Shipped 0.9.0 (with some delay)
